httpbp/headers.go

```diff
@@ -0,0 +1,117 @@
+package httpbp
+
+import (
+	"context"
+	"net/http"
+)
+
+type headerContextKey int
+
+const (
+	// HeaderEdgeContext is the key for the raw edge request context
+	HeaderEdgeContext headerContextKey = iota
+
+	// HeaderSpanTraceID is the header for the trace ID passed by the caller
+	HeaderSpanTraceID
+
+	// HeaderSpanParentID is the header for the parent ID passed by the caller
+	HeaderSpanParentID
+
+	// HeaderSpanID is the header for the span ID passed by the caller
+	HeaderSpanID
+
+	// HeaderSpanFlags is the header for the span flags passed by the caller
+	HeaderSpanFlags
+
+	// HeaderSpanSampled is the header for the sampled flag passed by the caller
+	HeaderSpanSampled
+)
+
+// GetHeader returns the HTTP header stored on the context at key.
+func GetHeader(ctx context.Context, key headerContextKey) (string, bool) {
```

Comment:
Using an unexported type (`headerContextKey`) in an exported function (`GetHeader`) won't work as the caller has no way to provide that arg.

We also should not export the context key type, as that could lead to context key collisions (someone use the exported type to set context value and bypass our logic, probably accidentally). See the doc on https://godoc.org/context#WithValue for more discussion.

So we probably should have a set of internal type/values for context key, and a set of exported enum for read-only access.